cmd/drone-server/server.go

```diff
@@ -542,34 +560,9 @@ func server(c *cli.Context) error {
 
 	// start the grpc server
 	g.Go(func() error {
-
-		lis, err := net.Listen("tcp", ":9000")
-		if err != nil {
-			logrus.Error(err)
-			return err
-		}
-		auther := &authorizer{
-			password: c.String("agent-secret"),
-		}
-		s := grpc.NewServer(
-			grpc.StreamInterceptor(auther.streamInterceptor),
-			grpc.UnaryInterceptor(auther.unaryIntercaptor),
-			grpc.KeepaliveEnforcementPolicy(keepalive.EnforcementPolicy{
-				MinTime: c.Duration("keepalive-min-time"),
-			}),
-		)
-		ss := new(droneserver.DroneServer)
-		ss.Queue = droneserver.Config.Services.Queue
-		ss.Logger = droneserver.Config.Services.Logs
-		ss.Pubsub = droneserver.Config.Services.Pubsub
-		ss.Remote = remote_
-		ss.Store = store_
-		ss.Host = droneserver.Config.Server.Host
-		proto.RegisterDroneServer(s, ss)
-
-		err = s.Serve(lis)
+		err := serveGrpc(c, remote_, store_)
 		if err != nil {
-			logrus.Error(err)
```

Comment:
Changed this from `Error` to `Fatal` as if invalid tls config was provided the server would start and hang and **not** return the error from TLS.

I think this is because of the errorgroup setup.  The grpc server and http servers are started in `g.Go` calls then later `g.Wait()` is called.  If the TLS config wants to return an error but `g.Wait()` is waiting for http server to finish which it never does.  So this change forces Drone to fail and returns any TLS config errors encountered.